sources/packages/api/src/services/application/application.service.ts

```diff
@@ -214,10 +214,17 @@ export class ApplicationService extends RecordDataModelService<Application> {
       [],
       associatedFiles,
     );
-    application.creator = auditUser;
-    application.studentAssessments = [originalAssessment];
-    application.currentAssessment = originalAssessment;
-    await this.repo.save([application, newApplication]);
+    await this.connection.transaction(async (transactionalEntityManager) => {
+      await transactionalEntityManager
+        .getRepository(Application)
+        .save(newApplication);
+      newApplication.creator = auditUser;
+      newApplication.studentAssessments = [originalAssessment];
+      newApplication.currentAssessment = originalAssessment;
+      await transactionalEntityManager
```

Comment:
Please add a comment that explains why the first application is being saved before on line 220 and later in line 226.